.github/workflows/ci.yml

```diff
@@ -0,0 +1,123 @@
+name: CI
+
+on:
+  pull_request:
+    types:
+      - opened
+      - synchronize
+
+env:
+  R4_RESOURCE_APIS_PATTERN: r4/resource-apis/*/*
+  R4_PATTERN: r4/*/*
+
+jobs:
+  setup:
+    runs-on: ubuntu-latest
+    outputs:
+      project-matrix: ${{ steps.unique-project-paths.outputs.BALLERINA_PROJECT_PATHS }}
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Extract Unique Ballerina Project Paths from Changed Files
+        id: unique-project-paths
+        run: |
+          # Get changed files of PR from github API
+          PR_DETAILS=$(curl -sSL -H "Authorization: Bearer ${{ secrets.GITHUB_TOKEN }}" \
```

Comment:
Do we need authorization here? Our repos are public so github apis may not need it.

Reply:
Ack. Will remove and update the PR